File: tiny-interpreter/cmd.ts
import { createInterface } from 'readline';
import Script from "./src/script";

const script = new Script();
Script.verbose = false;

const readline = createInterface({
  input: process.stdin,
  output: process.stdout,
  prompt: '>'
})
readline.prompt();

readline.on('line', (line) => {
  const lineCode = line.trim();
  if (lineCode == "exit()") {
    console.log('再见!');
    process.exit(0);
    return;
  }
  if (lineCode == "context") {
    console.log(script.kc);
    return;
  }

  script.process(lineCode);
  // prompt();
}).on('close', () => {
  console.log('再见!');
  process.exit(0);
});

File: cpp-tiny-interpreter/cmd.cpp
#include "iostream"
#include "./src/script.h"
using namespace std;
int main()
{
  Script *script = new Script();
  script->verbose = true;
  string inputCodeLine = "";
  cout << ">";
  while (getline(cin, inputCodeLine))
  {
    if (inputCodeLine.find("exit") != string::npos)
      return 0;
    script->process(inputCodeLine);
    cout << ">";
  }
  return 0;
}